includes/wc-stock-functions.php

```diff
@@ -307,12 +307,15 @@ function wc_get_held_stock_quantity( $product, $exclude_order_id = 0 ) {
 			"
 			SELECT SUM( order_item_meta.meta_value ) AS held_qty
 			FROM {$wpdb->posts} AS posts
+			LEFT JOIN {$wpdb->postmeta} as postmeta ON posts.ID = postmeta.post_id
 			LEFT JOIN {$wpdb->prefix}woocommerce_order_items as order_items ON posts.ID = order_items.order_id
 			LEFT JOIN {$wpdb->prefix}woocommerce_order_itemmeta as order_item_meta ON order_items.order_item_id = order_item_meta.order_item_id
 			LEFT JOIN {$wpdb->prefix}woocommerce_order_itemmeta as order_item_meta2 ON order_items.order_item_id = order_item_meta2.order_item_id
 			WHERE 	order_item_meta.meta_key    = '_qty'
 			AND 	order_item_meta2.meta_key   = %s
 			AND 	order_item_meta2.meta_value = %d
+			AND		postmeta.meta_key			= '_created_via'
+			AND		postmeta.meta_value			= 'checkout'
```

Comment:
Note that this is now hard coded, but `wc_cancel_unpaid_orders()` allows a filter called `woocommerce_cancel_unpaid_order` that allows stop cancel order created by other means.
Also the plugins that you report could use the same filter to properly call unpaid orders.
Since we are changing this query, we should also allow a filter now.

Reply:
The test for `get_created_via()==='checkout'` and the filter to control whether a given order gets cancelled are orthogonal, added two years apart (commit c19f2e5 in 2015 and 97ae238 in 2013, respectively).

I don't believe the _same_ undocumented filter (which can change false to true or vice versa!?) should be applied here. Nor do I see a motivation for a _new_ filter; what value would it provide, to whom? Perhaps @mikejolley or @maxrice can chime in.

Not sure what you mean by "properly call unpaid orders". The problem I had at first was from core WC, not the plugin, and could not be solved by the plugin using that filter. (On my site, the SUMO balance-due orders are _meant_ to hang around in pending state, perhaps for months, until the user pays the balance.)

The core problem IMO is inconsistency within WC:
- in `wc_cancel_unpaid_orders()`, whose purpose is to free **held stock**, only created-via-checkout orders are canceled, but
- in `wc_get_held_stock_quantity()`, the calculation of **held stock** for an item was not similarly counting only created-via-checkout orders.

That's all I want to fix in this PR.

Reply:
@claudiosanches Any more thoughts? I do not know what kind of filter you have in mind, and would appreciate a response to my questions above.

Reply:
Sorry, but I don't see a problem in including this new filter, since the query should be strict to just one result of `created_via`.
What is the problem for including this new filter if someone wants to change this query?
I haven't say that there's any problem with this PR, I just ask for you include a filter...

Reply:
I am fine with that but what should the filter do? I have very little experience hacking WP/WC. Looking for a little more guidance.

Reply:
@claudiosanches Waiting on your response...

Reply:
It's fine, I'll include a filter later on it if necessary. Thanks for your help!.